examples/test/qore/operators/operators.qtest

```diff
@@ -894,4 +895,191 @@ class Test inherits QUnit::Test {
         assertEq(NOTHING, True * <ab>);
         assertEq(NOTHING, NULL * False);
     }
+
+    rangeAndSlice() {
+        assertEq((3,4,5,6,7), 3..7);
+        assertEq((-2,-1,0,1,2,3,4,5,6), (1-3)..(2*3));
+        assertEq((6,5,4,3,2), 6..2);
+
+        list l = (7,3,5,1,8,7,2);
+        assertEq((1,3,8), l[3,1,4]);
+        assertEq((5,1,8,7), l[2..5]);
+        assertEq((7,8,1,5), l[5..2]);
+        assertEq((8,7,2), l[4..]);
+        assertEq((8,7,2), l[4..12]);
```

Comment:
Shouldn't this return a list with a few NOTHING values?

Reply:
not according to the current definition; the reason is that extra invalid elements can only appear at the end of a range, where they would all be `NOTHING`, not in the middle as with a list of arbitrary index positions, so therefore the original implementation was left as-is

Reply:
Intuitively, I would expect that the following expressions will return the same thing no matter the contents of `a`:
```
a[2..4]
a[(2..4)]
a[2, 3, 4]
a[(2,3,4)]

list x = 2..4;
a[x]
list x = (2..4);
a[x]
list x = (2,3,4);
a[x]
```
But this does not seem that that's the case.

Reply:
The `..` operator should be just a shortcut to the integer list with the right values...

Reply:
@omusil24 that's not how it's been implemented, but I think you are right about the consistency issues - it would be more consistent for ranges to return `NOTHING` also for invalid elements; when using a one-sided range `[5..]` then it would return up to the end of the list|string|binary.

Reply:
@omusil24 done - implementation is now consistent